src/mongoose_backend.erl

```diff
@@ -20,12 +20,13 @@
 -type function_name() :: atom().
 -type main_module() :: module().
 -type backend_module() :: module().
+-type host_type_or_global() :: host_type_or_global() | global.
 
--spec init_per_host_type(HostType :: mongooseim:host_type(),
+-spec init(HostType :: host_type_or_global(),
```

Comment:
Also, the name of the variable is by now misleading, this is not a host_type any more, so just like it is done for `gen_hook`, all these `HostType` vars can be renamed to `Tag` or something like that.